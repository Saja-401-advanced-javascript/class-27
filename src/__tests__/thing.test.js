import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Counter from '../components/counter/counter.js';

Enzyme.configure({ adapter: new Adapter() });

describe('Testing our Counter component', () => {

    it('exists at the start of the app', () => {
        let app = shallow(<Counter />);
        expect(app.find('h3').exists()).toBeTruthy();
    })

    it('Counter renders', () => {
        let app = mount(<Counter />);
        let button = app.find('button');
        expect(button).toBeDefined();
        expect(app.state('counter')).toBe(0);
    });

    it('render correctly to the DOM', () => {
        const renderTree = renderer.create(<Counter />).toJSON();
        expect(renderTree).toMatchSnapshot();
    });

    it('Increase Counter value by one on Click', () => {
     
        let app = mount(<Counter />);
        let button = app.find('#one');      
        button.simulate('click');
        expect(app.state('counter')).toBe(1);
    });

    it('Decrease Counter value by one on Click', () => {
     
        let app = mount(<Counter />);
        let button = app.find('#two');      
        button.simulate('click');
        expect(app.state('counter')).toBe(-1);
    });
});


