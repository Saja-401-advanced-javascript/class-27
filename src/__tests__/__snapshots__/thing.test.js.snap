// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing our Counter component Assert that state is being transferred to the DOM 1`] = `
<section>
  <h3
    style={
      Object {
        "color": "green",
      }
    }
  >
    0
  </h3>
  <button
    id="one"
    onClick={[Function]}
  >
    click to increase counter
  </button>
  <button
    id="two"
    onClick={[Function]}
  >
    click to decrease counter
  </button>
</section>
`;

exports[`Testing our Counter component render correctly to the DOM 1`] = `
<section>
  <h3
    style={
      Object {
        "color": "green",
      }
    }
  >
    0
  </h3>
  <button
    id="one"
    onClick={[Function]}
  >
    click to increase counter
  </button>
  <button
    id="two"
    onClick={[Function]}
  >
    click to decrease counter
  </button>
</section>
`;
